src/api/apiPcl.cpp

```diff
@@ -12,54 +12,81 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
+#include <string>
+
+#include <pcl/point_cloud.h>
+#include <pcl/point_types.h>
+#include <pcl/io/pcd_io.h>
+
 #include <rgl/api/extensions/pcl.h>
 #include <api/apiCommon.hpp>
+#include <graph/NodesCore.hpp>
 #include <graph/NodesPcl.hpp>
 
+#include <RGLExceptions.hpp>
+#include <RGLFields.hpp>
+
 extern "C" {
 
 RGL_API rgl_status_t
-rgl_node_points_downsample(rgl_node_t* node, float leaf_size_x, float leaf_size_y, float leaf_size_z)
+rgl_graph_write_pcd_file(rgl_node_t node, const char* file_path)
 {
 	auto status = rglSafeCall([&]() {
-		RGL_API_LOG("rgl_node_points_downsample(node={}, leaf=({}, {}, {}))", repr(node), leaf_size_x, leaf_size_y, leaf_size_z);
+		RGL_API_LOG("rgl_graph_write_pcd_file(node={}, file={})", repr(node), file_path);
+		CHECK_ARG(file_path != nullptr);
+		CHECK_ARG(file_path[0] != '\0');
 
-		createOrUpdateNode<DownSamplePointsNode>(node, Vec3f{leaf_size_x, leaf_size_y, leaf_size_z});
+		auto pointCloudNode = Node::validatePtr<IPointsNode>(node);
+
+		if (!pointCloudNode->hasField(XYZ_F32)) {
+			throw InvalidAPIObject(fmt::format("Saving PCD file {} failed - requested node hasn't field XYZ.", file_path));
```

Comment:
`node hasn't field`
Same case as in the last PR. Correct but misleading use of language.